src/main/java/com/terraformersmc/terrestria/feature/tree/foliageplacers/PredictiveSpruceFoliagePlacer.java

```diff
@@ -55,12 +54,9 @@ protected void generateSquare(ModifiableTestableWorld world, Random random, Tree
 					mutable.set(blockPos, offsetX, offsetY, offsetZ);
 					if (TreeFeature.canReplace(world, mutable)) {
 						actualDistance = calculateActualDistance(offsetX, offsetY, offsetZ, giantTrunk);
-						BlockState baseState = config.leavesProvider.getBlockState(random, mutable);
+						BlockState baseState = config.foliageProvider.getBlockState(random, mutable);
 
-						world.setBlockState(mutable, withDistance(baseState, actualDistance), 19);
-
-						blockBox.encompass(new BlockBox(mutable, mutable));
-						set.add(mutable.toImmutable());
+						replacer.accept(mutable.toImmutable(), withDistance(baseState, actualDistance));
```

Comment:
This is likely going to cause problems with leaf decay on our ExtendedLeavesBlock stuff for the redwoods and hemlock

Reply:
In which way? I have done some tests, and everything seems to work fine (except that decaying leaves drop themselves, but that's another issue) 🤔

Reply:
Mentioned it in the other PR, but I was talking about the removed lines and the large trees with longer decay distances